cmake/target/install.cmake

```diff
@@ -43,12 +43,16 @@ function(install_add_deployment_target MODULE TARGET SOURCES DEPENDENCIES FULL_D
     set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE)
     _install_real_helper(INSTALL_DEPENDENCIES "${FULL_DEPENDENCIES}")
     install(TARGETS ${MODULE} ${INSTALL_DEPENDENCIES}
-            RUNTIME DESTINATION ${TOOLCHAIN_NAME}/bin
-            LIBRARY DESTINATION ${TOOLCHAIN_NAME}/lib
-            ARCHIVE DESTINATION ${TOOLCHAIN_NAME}/lib/static)
-    get_property(DICTIONARY GLOBAL PROPERTY "${PROJECT_NAME}_FPRIME_DICTIONARY_FILE")
-    install(FILES ${DICTIONARY} DESTINATION ${TOOLCHAIN_NAME}/dict)
-    add_custom_command(TARGET "${MODULE}" POST_BUILD COMMAND "${CMAKE_COMMAND}" --build "${CMAKE_BINARY_DIR}" --target install)
+            RUNTIME DESTINATION ${TOOLCHAIN_NAME}/${MODULE}/bin
+            COMPONENT ${MODULE}
+            LIBRARY DESTINATION ${TOOLCHAIN_NAME}/${MODULE}/lib
+            COMPONENT ${MODULE}
+            ARCHIVE DESTINATION ${TOOLCHAIN_NAME}/${MODULE}/lib/static
+            COMPONENT ${MODULE}
+    )
+    install(FILES ${FPRIME_CURRENT_DICTIONARY_FILE} DESTINATION ${TOOLCHAIN_NAME}/${MODULE}/dict COMPONENT ${MODULE})
+    add_custom_command(TARGET "${MODULE}" POST_BUILD COMMAND "${CMAKE_COMMAND}"
+            -DCMAKE_INSTALL_COMPONENT=${MODULE} -P${CMAKE_BINARY_DIR}/cmake_install.cmake)
```

Comment:
I'm opening #2031 to fix an issue introduced here:
```
CMake Error: No script specified for argument -P
gmake[3]: *** [CMakeFiles/Ref.dir/build.make:169: bin/Linux/Ref] Error 1
gmake[3]: *** Deleting file 'bin/Linux/Ref'
gmake[2]: *** [CMakeFiles/Makefile2:2352: CMakeFiles/Ref.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:2359: CMakeFiles/Ref.dir/rule] Error 2
gmake: *** [Makefile:184: Ref] Error 2
[ERROR] CMake erred with return code 2
root@2d500cea2ecc:/home/ptl# cmake --version
cmake version 3.18.4
```

Verification that this is the offending line:
```
root@2d500cea2ecc:/home/ptl/Ref/build-fprime-automatic-native# /usr/bin/cmake -DCMAKE_INSTALL_COMPONENT=Ref -P/home/ptl/Ref/build-fprime-automatic-native/cmake_install.cmake
CMake Error: No script specified for argument -P
```

With a space after `-P` builds work for my version of CMake.